602-spring-data-rest/src/test/java/org/acme/spring/data/rest/BookRepositoryTest.java

```diff
@@ -0,0 +1,160 @@
+package org.acme.spring.data.rest;
+
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import org.acme.spring.data.rest.containers.PostgreSqlDatabaseTestResource;
+import org.junit.Assert;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+import javax.inject.Inject;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.core.IsNot.not;
+
+@QuarkusTest
+@QuarkusTestResource(PostgreSqlDatabaseTestResource.class)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
```

Comment:
The tests are not independent from each other. Can you adjust the test cases accordingly? For example, we could create the job in the beforeEach stage and then proceed with the test cases.

Reply:
Can you be more specific, please. I didn't get why they are dependent on each other. Is it about @QuarkusTestResource(PostgreSqlDatabaseTestResource.class) being used in each test?

Reply:
It's not about using `@QuarkusTestResource`, but, as an example, if you try to run only the second test `testRelationsBetweenEntities`, it will fail because it needs the first test to be run `testListAllBooks`. The same happens with the third test `testSorting`. This is what I meant with the tests are not independent from each other. 

This can be easily solved by either adding the common data as part of the `beforeEach` test stage, or by creating the data needed in each test or by adding the data in the `import.sql`.

Reply:
got you